src/services/database.ts

```diff
@@ -68,6 +68,89 @@ export class DatabaseService {
     return await db.bookmarks.where('unread').equals(1).toArray();
   }
 
+  static async getBookmarksPaginated(filter: 'all' | 'unread' | 'archived', page: number, pageSize: number): Promise<LocalBookmark[]> {
+    const offset = (page - 1) * pageSize;
+    
+    if (filter === 'unread') {
+      return await db.bookmarks
+        .where('unread').equals(1)
+        .and(bookmark => !bookmark.is_archived)
+        .offset(offset)
+        .limit(pageSize)
+        .toArray();
+    } else if (filter === 'archived') {
+      return await db.bookmarks
+        .where('is_archived').equals(1)
+        .offset(offset)
+        .limit(pageSize)
+        .toArray();
+    } else {
+      // 'all' filter shows only unarchived bookmarks
+      return await db.bookmarks
+        .orderBy('date_added')
+        .reverse()
```

Comment:
Shouldn't the 'unread' and 'archived' filters also use this ordering? Combine the 3 if statements so that they configure the shared query properties in one place and then branch for the query properties that differ between the 3 filters.